### ISSUE TITLE
Can't compile using latest NPM version

### ISSUE THREAD

**Issue by Synergyst** [google/shaka-player#1758]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**:
Yes
**What version of Shaka Player are you using?**:
Latest from GitHub at the time of this post.
**Can you reproduce the issue with our latest release version?**:
I am using the latest..
**Can you reproduce the issue with the latest code from `master`?**:
I am using the latest from master..
**Are you using the demo app or your own custom app?**:
I am compiling..
**If custom app, can you reproduce the issue using our demo app?**:
N/A
**What browser and OS are you using?**:
Debian 9
**What are the manifest and license server URIs?**:
N/A
**What did you do?**
I tried compiling Shaka following these instructions: https://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html
**What did you expect to happen?**
I expected Shaka to compile.
**What actually happened?**
Shaka did not compile..

```
root@live:/usr/local/nginx/html/shaka-player/build# PRINT_ARGUMENTS=1 python2.7 all.py
[INFO] Generating Closure dependencies...
[INFO] /usr/bin/python2.7 third_party/closure/deps/depswriter.py "--root_with_prefix=lib ../../../lib" "--root_with_prefix=ui ../../../ui" "--root_with_prefix=third_party/closure ../../../third_party/closure" "--root_with_prefix=third_party/language-mapping-list ../../../third_party/language-mapping-list"
[INFO] npm -v
Traceback (most recent call last):
  File "all.py", line 122, in <module>
    shakaBuildHelpers.run_main(main)
  File "/usr/local/nginx/html/shaka-player/build/shakaBuildHelpers.py", line 312, in run_main
    sys.exit(main(sys.argv[1:]))
  File "all.py", line 70, in main
    if check.main(check_args) != 0:
  File "/usr/local/nginx/html/shaka-player/build/check.py", line 157, in main
    if not shakaBuildHelpers.update_node_modules():
  File "/usr/local/nginx/html/shaka-player/build/shakaBuildHelpers.py", line 278, in update_node_modules
    if _parse_version(version) < _parse_version('1.3.12'):
  File "/usr/local/nginx/html/shaka-player/build/shakaBuildHelpers.py", line 59, in _parse_version
    return tuple([int(i) for i in version.split('.')])
ValueError: invalid literal for int() with base 10: '0-next'
```

**Comment by TheModMaker** [comment]
It looks like we can't handle the version number of your npm install.  Are you using a pre-release or compiled from source version of npm?  What is the output of `npm --version`?  Can you try with a release version of npm (e.g. 6.4.1).

**Comment by Synergyst** [comment]
I am using the latest version from Node's site.
Should I downgrade to their LTS version?

```
# npm --version
6.5.0-next.0
```

**Comment by Synergyst** [comment]
Alright..

It seems to be working now.

I'll close this issue here in a bit **if** it compiles correctly.

**Comment by TheModMaker** [comment]
We can fix it to work with the latest NPM version.  But at least there's a workaround in the meantime.

**Comment by Synergyst** [comment]
Resolved temporarily (until changes are commited) by using the following version of NodeJS: v10.15.0 (https://nodejs.org/dist/v10.15.0/node-v10.15.0-linux-x64.tar.xz)

**Comment by joeyparrish** [comment]
Should be easy to fix.

**Comment by joeyparrish** [comment]
Fix cherry-picked for v2.4.6.
